packages/next/build/webpack-config.ts

```diff
@@ -1577,61 +1577,74 @@ export default async function getBaseWebpackConfig(
-                      fallback: {
-                        assert: require.resolve('next/dist/compiled/assert'),
-                        buffer: require.resolve('next/dist/compiled/buffer/'),
-                        constants: require.resolve(
-                          'next/dist/compiled/constants-browserify'
-                        ),
-                        crypto: require.resolve(
-                          'next/dist/compiled/crypto-browserify'
-                        ),
-                        domain: require.resolve(
-                          'next/dist/compiled/domain-browser'
-                        ),
-                        http: require.resolve('next/dist/compiled/stream-http'),
-                        https: require.resolve(
-                          'next/dist/compiled/https-browserify'
-                        ),
-                        os: require.resolve('next/dist/compiled/os-browserify'),
-                        path: require.resolve(
-                          'next/dist/compiled/path-browserify'
-                        ),
-                        punycode: require.resolve(
-                          'next/dist/compiled/punycode'
-                        ),
-                        process: require.resolve('./polyfills/process'),
-                        // Handled in separate alias
-                        querystring: require.resolve(
-                          'next/dist/compiled/querystring-es3'
-                        ),
-                        stream: require.resolve(
-                          'next/dist/compiled/stream-browserify'
-                        ),
-                        string_decoder: require.resolve(
-                          'next/dist/compiled/string_decoder'
-                        ),
-                        sys: require.resolve('next/dist/compiled/util/'),
-                        timers: require.resolve(
-                          'next/dist/compiled/timers-browserify'
-                        ),
-                        tty: require.resolve(
-                          'next/dist/compiled/tty-browserify'
-                        ),
-                        // Handled in separate alias
-                        // url: require.resolve('url/'),
-                        util: require.resolve('next/dist/compiled/util/'),
-                        vm: require.resolve('next/dist/compiled/vm-browserify'),
-                        zlib: require.resolve(
-                          'next/dist/compiled/browserify-zlib'
-                        ),
-                        events: require.resolve('next/dist/compiled/events/'),
-                        setImmediate: require.resolve(
-                          'next/dist/compiled/setimmediate'
-                        ),
-                      },
+                      fallback:
+                        config.experimental.fallbackOldPolyfills === false
```

Comment:
feel like we don't need `old` as descriptor here? the flag remove all the fallback polyfills, more like enabling or disabling